### ISSUE TITLE
Autocomplete list doesnt reset in MatChipList after adding a Chip

### ISSUE THREAD

**Issue by CurtisDS** [angular/material2#11291]
#### Bug, feature request, or proposal:
Bug

#### What is the expected behavior?
After using autocomplete to add a chip the next time the user clicks on the input field it should show the full list of options again.

#### What is the current behavior?
After using autocomplete to add a chip the next time the user clicks on the input field it only shows the options that were shown previously. In order to see all options the user needs to press space and then backspace.

#### What are the steps to reproduce?
Go to the autocomplete example from on [material.angular.io](https://material.angular.io/components/chips/overview#chip-input). Or to this [stackblitz](https://stackblitz.com/angular/krakmngdppp?file=app%2Fchips-autocomplete-example.ts).

Steps:
Click in the input box (all options will be shown).
Select an option, for example `Orange`
Either press down on the keyboard or remove focus from the control by click outside and refocus it to open the drop down

Result:
Only `Orange` is shown as an option even though no text is in the input field to filter the list.

#### What is the use-case or motivation for changing an existing behavior?
It seems like a poor UI choice. It gives inconsistent results for the same actions.

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
Material 6.0.1

**Comment by CurtisDS** [comment]
I forked the example and fixed it: https://stackblitz.com/edit/angular-xp9pbd?file=app/chips-autocomplete-example.ts

**Comment by jelbourn** [comment]
@CurtisDS did you want to send a PR?

**Comment by CurtisDS** [comment]
@jelbourn I can try. I've never done that before.

**Comment by CurtisDS** [comment]
@jelbourn hmmmm it looks like its complicated. lol. I'd rather let someone else do it.

**Comment by julianobrasil** [comment]
Also, the example does not allow chip removal (it worked fine on stackblitz, but didn't work at all on docs site => Windows 10 + Chrome):

![image](https://media.giphy.com/media/3rSPkvic35Eekjz0x8/giphy.gif)

**Comment by CurtisDS** [comment]
@julianobrasil 
Yes I submitted that issue here https://github.com/angular/material2/issues/11290 (about not being able to remove the chip)
In the stackblitz I fixed both issues, so you should be able to remove chips.

**Comment by julianobrasil** [comment]
Ok, I'll submit a PR with your fix, @CurtisDS.
